src/lib/flash-helper.js

```diff
@@ -38,29 +38,55 @@ async function flashFiles({ device, flashSteps, resetAfterFlash = true, ui }) {
 }
 
 async function _flashDeviceInNormalMode(device, data, { name, progress } = {}) {
-	if (device.isInDfuMode) {
-		// put device in normal mode
-		if (progress) {
-			progress({ event: 'switch-mode', mode: 'normal' });
-		}
-		device = await usbUtils.reopenInNormalMode(device, { reset: true });
-	}
-	// put the device in listening mode to prevent cloud connection
-	try {
-		await device.enterListeningMode();
-	} catch (error) {
-		// ignore if the device is already in listening mode
-	}
-
 	// flash the file in normal mode
 	if (progress) {
 		progress({ event: 'flash-file', filename: name });
 	}
-	await device.updateFirmware(data, { progress, timeout: FLASH_TIMEOUT });
 
-	// wait for the device to apply the firmware
-	await delay(FLASH_APPLY_DELAY);
-	device = await usbUtils.reopenInNormalMode(device, { reset: false });
+	const start = Date.now();
+	while (Date.now() - start < FLASH_TIMEOUT) {
```

Comment:
This is the retry section. I'm not sure on doing it by using an artificial timeout or maybe should be better use a retries number.

Reply:
In both: `device-os-flash-util` and `web-setup` we use retry number